model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro.py

```diff
@@ -465,6 +642,7 @@ def init(
             vertical_params,
             edge_geometry,
             owner_mask,
+            backend=self._backend,
```

Comment:
you can remove the `backend` argument here, as you have access to the backend through the instance variable `self._backend`

Reply:
yes good point

Reply:
we need this if we want to call VelocityAdvection outside of solveNonHydro. Is that ever the case?